conjure-java-jackson-serialization/src/main/java/com/palantir/conjure/java/serialization/LenientLongModule.java

```
@@ -0,0 +1,56 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.serialization;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import java.io.IOException;
+
+/**
+ * Provides support for the {@link Long} deserialization from JSON string and numeric values regardless of
+ * <pre>MapperFeature.ALLOW_COERCION_OF_SCALARS</pre> configuration.
+ */
+final class LenientLongModule extends SimpleModule {
+
+    LenientLongModule() {
+        super("lenient long");
+        // Register to both Long.TYPE and Long.class
+        this.addDeserializer(long.class, new LongAsStringDeserializer())
+                .addDeserializer(Long.class, new LongAsStringDeserializer());
+    }
+
+    private static final class LongAsStringDeserializer extends StdDeserializer<Long> {
+
+        private LongAsStringDeserializer() {
+            super(Long.TYPE);
+        }
+
+        @Override
+        public Long deserialize(JsonParser jsonParser, DeserializationContext ctxt) throws IOException {
+            switch (jsonParser.currentToken()) {
+                case VALUE_NUMBER_INT:
+                    // Lenient implementation for compatibility with existing clients
+                    return jsonParser.getLongValue();
+                case VALUE_STRING:
+                    return Long.valueOf(jsonParser.getValueAsString());
+            }
```

Do we need to handle `VALUE_NULL`?

---

I suppose it's possible for java code to expect a boxed `Long` in a nullable place, though in an ideal world we would expect `OptionalLong`. Will update.

---

Should we not fail if a long is expected but the value is absent?

---

We should, but that's a separate concern from this parsing function. It's possible that existing human generated objects are expected to be deserialized as `@Nullable Long value` which we shouldn't cause to throw in this change.